spec/integration/outputs/retry_spec.rb

```diff
@@ -11,146 +11,143 @@
   let(:max_retries) { 3 }
 
   def mock_actions_with_response(*resp)
-    LogStash::Outputs::Elasticsearch::Protocols::HTTPClient
+    LogStash::Outputs::Elasticsearch::Protocols::TransportClient
```

Comment:
2 things.

1. I don't fully understand why, but mocking both Node and Transport Clients caused problems to me in the past. Since Transport client is of type NodeClient, mocking the NodeClient sufficed.

2. we might as well update to the newer convention in rspec for any_instance mocking like so:

```
allow_any_instance_of(LogStash::Outputs::Elasticsearch::Protocols::NodeClient).to receive(:bulk).and_return(*resp)
```